R/batchmark.R

```diff
@@ -0,0 +1,175 @@
+#' @title Run machine learning benchmarks as distributed experiments.
+#'
+#' @description
+#' This function is a very parallel version of \code{\link{benchmark}}.
+#' Experiments are created in the provided registry for each combination of
+#' learners, tasks and resamplings. The experiments are then stored in a registry and the
+#' runs can be started via \code{\link[batchtools]{submitJobs}]}. A job is one train/test split
+#' of the outer resampling. In case of nested resampling (e.g. with \code{\link{makeTuneWrapper}}),
+#' each job is a full run of inner resampling, which can be parallelized in a second step
+#' with \pkg{ParallelMap}. For details on the usage and support backends have
+#' a look at the batchtools tutorial page:
+#' \url{https://github.com/mllg/batchtools}.
+#'
+#' The general workflow with \code{batchmark} looks like that:
+#' \enumerate{
+#' \item{1.}{\code{batchtools::createExperimentRegistry()}}
+#' \item{2.}{\code{batchmark(...)}}
+#' \item{3.}{\code{batchtools:submitJobs()}}
+#' \item{4.}{wait until the jobs are finished...}
+#' \item{5.}{\code{reduceBatchtoolsResult()}}
+#' }
+#' @inheritParams benchmark
+#' @param data.ids [\code{integer}]\cr
+#'   Dataset IDs to download from Open-ML. Default is none.
+#' @param resamplings [(list of) \code{\link{ResampleDesc}}]\cr
+#'   Resampling strategy for each tasks.
+#'   If only one is provided, it will be replicated to match the number of tasks.
+#'   If missing, a 10-fold cross validation is used.
+#' @param reg [\code{\link[batchtools]{Registry}}]\cr
+#'   Registry, created by \code{\link[batchtools]{makeExperimentRegistry}}. If not explicitly passed,
+#'   uses the last created registry.
+#' @return [\code{data.table}]. Generated job ids are stored in the column \dQuote{job.id}.
+#' @export
+#' @family benchmark
+batchmark = function(learners, tasks, data.ids, resamplings, measures, models = TRUE, reg = batchtools::getDefaultRegistry()) {
+
+  requirePackages("batchtools", why = "batchmark", default.method = "load")
+
+  learners = ensureBenchmarkLearners(learners)
+  learner.ids = vcapply(learners, getLearnerId)
+  names(learners) = learner.ids
+
+  if (!missing(tasks)) {
```

Comment:
Without data.ids, this is pretty straightforward. You need to pass tasks, thus you can use checkmate here and assert a list of tasks (or a single task). Probably there is already a helper in mlr to do this.